aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -150,6 +214,272 @@ module aptos_framework::coin {
+        let mint_ref_opt = &mut borrow_global_mut<PairedFungibleAssetRefs>(metadata_addr).mint_ref_opt;
+        option::fill(mint_ref_opt, mint_ref);
+    }
+
+    #[view]
+    /// Check whether `TransferRef` has not been taken.
+    public fun paired_transfer_ref_exists<CoinType>(): bool acquires CoinConversionMap, PairedFungibleAssetRefs {
+        let metadata = assert_paired_metadata_exists<CoinType>();
+        let metadata_addr = object_address(&metadata);
+        assert!(exists<PairedFungibleAssetRefs>(metadata_addr), error::internal(EPAIRED_FUNGIBLE_ASSET_REFS_NOT_FOUND));
+        option::is_some(&borrow_global<PairedFungibleAssetRefs>(metadata_addr).transfer_ref_opt)
+    }
+
+    /// Get the TransferRef of paired fungible asset of a coin type from `FreezeCapability`.
+    public fun get_paired_transfer_ref<CoinType>(
+        _: &FreezeCapability<CoinType>
+    ): (TransferRef, TransferRefReceipt) acquires CoinConversionMap, PairedFungibleAssetRefs {
+        let metadata = assert_paired_metadata_exists<CoinType>();
+        let metadata_addr = object_address(&metadata);
+        assert!(exists<PairedFungibleAssetRefs>(metadata_addr), error::internal(EPAIRED_FUNGIBLE_ASSET_REFS_NOT_FOUND));
+        let transfer_ref_opt = &mut borrow_global_mut<PairedFungibleAssetRefs>(metadata_addr).transfer_ref_opt;
+        assert!(option::is_some(transfer_ref_opt), error::not_found(ETRANSFER_REF_NOT_FOUND));
+        (option::extract(transfer_ref_opt), TransferRefReceipt { metadata })
+    }
+
+    /// Return the `TransferRef` with the hot potato receipt.
+    public fun return_paired_transfer_ref(
+        transfer_ref: TransferRef,
+        receipt: TransferRefReceipt
+    ) acquires PairedFungibleAssetRefs {
+        let TransferRefReceipt { metadata } = receipt;
+        assert!(
+            fungible_asset::transfer_ref_metadata(&transfer_ref) == metadata,
+            error::invalid_argument(ETRANSFER_REF_RECEIPT_MISMATCH)
+        );
+        let metadata_addr = object_address(&metadata);
+        assert!(exists<PairedFungibleAssetRefs>(metadata_addr), error::internal(EPAIRED_FUNGIBLE_ASSET_REFS_NOT_FOUND));
+        let transfer_ref_opt = &mut borrow_global_mut<PairedFungibleAssetRefs>(metadata_addr).transfer_ref_opt;
+        option::fill(transfer_ref_opt, transfer_ref);
+    }
+
+    #[view]
+    /// Check whether `TransferRef` has not been taken.
+    public fun paired_burn_ref_exists<CoinType>(): bool acquires CoinConversionMap, PairedFungibleAssetRefs {
+        let metadata = assert_paired_metadata_exists<CoinType>();
+        let metadata_addr = object_address(&metadata);
+        assert!(exists<PairedFungibleAssetRefs>(metadata_addr), error::internal(EPAIRED_FUNGIBLE_ASSET_REFS_NOT_FOUND));
+        option::is_some(&borrow_global<PairedFungibleAssetRefs>(metadata_addr).burn_ref_opt)
+    }
+
+    /// Get the `BurnRef` of paired fungible asset of a coin type from `BurnCapability`.
+    public fun get_paired_burn_ref<CoinType>(
```

Comment:
What happens at the end of the migration? How do creators obtain Mint, Burn, Transfer Refs?

Reply:
This would need another AIP and API.